pkg/util/helm_helpers.go

```diff
@@ -22,96 +22,308 @@ under the License.
+	// 		if client.DependencyUpdate {
+	// 			man := &downloader.Manager{
+	// 				Out:              out,
+	// 				ChartPath:        cp,
+	// 				Keyring:          client.ChartPathOptions.Keyring,
+	// 				SkipUpdate:       false,
+	// 				Getters:          p,
+	// 				RepositoryConfig: settings.RepositoryConfig,
+	// 				RepositoryCache:  settings.RepositoryCache,
+	// 			}
+	// 			if err := man.Update(); err != nil {
+	// 				return nil, err
+	// 			}
+	// 		} else {
+	// 			return nil, err
+	// 		}
+	// 	}
+	// }
+
+	client := action.NewInstall(actionConfig)
+	if client.Version == "" && client.Devel {
+		client.Version = ">0.0.0-0"
+	}
+	client.ReleaseName = releaseName
+	client.DryRun = dryRun
+	_, err = client.Run(chart, vals) // deploy the chart into the namespace from the actionConfig
 	if err != nil {
-		return string(stdoutErr), fmt.Errorf("failed to run Helm command of args %+v with error %s", cmdArgs, err)
+		return fmt.Errorf("failed to Run NewInstall: %+v", err)
 	}
-	return string(stdoutErr), nil
+	return nil
 }
 
-func genHelm3Args(command string, args []string, setValuesMap map[string]string) []string {
-	helmArgs := append([]string{command}, args...)
-	for name, value := range setValuesMap {
-		helmArgs = append(helmArgs, "--set", fmt.Sprintf("%s=%s", name, value))
+// UpdateWithHelm3 uses the helm NewUpgrade action to update a resource in the cluster
+func UpdateWithHelm3(releaseName, namespace, chartURL string, vals map[string]interface{}, kubeConfig string) error {
+	actionConfig, err := CreateHelmActionConfiguration(kubeConfig, "", namespace)
+	if err != nil {
+		return err
+	}
+	if releaseExists := ReleaseExists(releaseName, namespace, kubeConfig); !releaseExists {
+		return fmt.Errorf("release '%s' does not exist", releaseName)
+	}
+
+	chart, err := LoadChart(chartURL, actionConfig)
+	if err != nil {
+		return fmt.Errorf("failed to load release at '%s' for updating: %s", chartURL, err)
+	}
+	validInstallableChart, err := isChartInstallable(chart)
+	if !validInstallableChart {
+		return fmt.Errorf("release at '%s' is not installable: %+v", chartURL, err)
+	}
+
+	client := action.NewUpgrade(actionConfig)
+	if client.Version == "" && client.Devel {
+		client.Version = ">0.0.0-0"
 	}
-	return helmArgs
+	client.ReuseValues = true                     // rememeber the values that have been set previously
+	_, err = client.Run(releaseName, chart, vals) // updates the release in the namespace from the actionConfig
+	if err != nil {
+		return fmt.Errorf("failed to Run NewUpgrade: %+v", err)
+	}
+	return nil
 }
 
-// HelmV3Exists returns true if it can find the helm binary and it is v3
-func HelmV3Exists() (bool, error) {
-	helmExists, err := HelmIsInPath()
+// TemplateWithHelm3 prints the kube manifest files for a resource
+func TemplateWithHelm3(releaseName, namespace, chartURL string, vals map[string]interface{}) error {
+	actionConfig, err := CreateHelmActionConfiguration("", "", namespace)
```

Comment:
Calling CreateHelmActionConfiguration, LoadChart and isChartInstallable is common in create, template, upgrade methods.. Can we create a function to do that?